owlkettle/widgetdef.nim

```diff
@@ -31,7 +31,9 @@ import bindings/gtk
 type
   Widget* = ref object of RootObj
     app*: Viewable
-  
+    hasStateRef*: bool
+    valStateRef*: StateRef
```

Comment:
If you use custom syntax, you can just have a single `stateRef: StateRef` field here.

Reply:
Given that stateRef is very much an optional field and a ref-type which might be nil, mind if I use `Option[StateRef]`?

Reply:
`nil` should already express the absence of a `StateRef`.